library/std/src/io/mod.rs

```diff
@@ -1418,6 +1418,25 @@ pub trait Write {
     /// The default implementation calls [`write`] with either the first nonempty
     /// buffer provided, or an empty one if none exists.
     ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::io::prelude::*;
+    /// use std::fs::File;
+    ///
+    /// fn main() -> std::io::Result<()> {
+    ///     let mut data1 = [1; 8];
+    ///     let mut data2 = [15; 8];
+    ///     let mut io_slice1 = IoSlice::new(&mut data1);
+    ///     let mut io_slice2 = IoSlice::new(&mut data2);
+    ///
+    ///     let mut buffer = File::create("foo.txt")?;
+    ///
+    ///     buffer.write_vectored(&[io_slice1, io_slice2])?;
```

Comment:
The `write` example contains a comment highlighting that it's not guaranteed to write all data. Either the same should be done here too or the example should handle the return value.

Reply:
This is a good point. I'll update the example.